src/domain/model/user/User.ts

```diff
@@ -73,7 +72,7 @@ export class User {
       role
         .permissions()
         .map((permission) => permission.name)
-        .forEach((permissionName) => permissions.add(permissionName));
+        .forEach((permission) => permissions.add(permission));
```

Comment:
For clarity, I would write this:
```
.permissions()
.map((permission) => permission.name)
.foreach((permissionName) => permissions.add(permissionName));
```

Or if you want to get real fancy:
```
.permissions()
.forEach(({ name }) => permissions.add(name));
```
Though I think the former is probably easier to read.

Reply:
I agree the former is a bit more readable. Still, loved what you did there with destructuring   :-)